src/main/java/se/liu/ida/hefquin/engine/queryplan/logical/LogicalPlanUtils.java

```diff
@@ -55,4 +54,15 @@ public void visit( final LogicalOpUnion op ) {
 		}
 	} // end of class SourceAssignmentChecker
 
+	public static String printStringOfFm( final FederationMember fm ) {
+
+		final DataRetrievalInterface intFace = fm.getInterface();
+		if ( intFace instanceof SPARQLEndpointInterface){
+			return "SPARQL endpoint at " + ((SPARQLEndpointInterface) intFace).getURL();
+		}
+		else {
+			return intFace.getClass().getName() + " server at TODO";
+		}
+	}
+
```

Comment:
I don't think this is needed. Instead, also for the `FederationMember` objects we can rely on their `toString()` method.